lib/xapi-stdext-date/date.ml

```diff
@@ -30,9 +30,14 @@ let rfc822_to_string x = x
 
 (* ==== ISO8601/RFC3339 ==== *)
 
-type iso8601 = Ptime.t
+type print_type = PrintLocal | PrintUTC
+(* we must store the print_type with iso8601 to handle the case where the local time zone is UTC *)
+type iso8601 = Ptime.date * Ptime.time * print_type
 
- let of_string x =
+let of_dt print_type dt = let (date, time) = dt in (date, time, print_type)
```

Comment:
This works, you can do the pattern matching earlier though:
`let of_dt print_type (date, time)`